### DISCUSSION TITLE
How to use controls from DotSpatial.Controls in WPF application?

### DISCUSSION THREAD

**Discussion by alsl521** [DotSpatial/DotSpatial#1568] [Q&A]
I'm new to programming and I want to use the controls in DotSpatial.Controls in a WPF application, but I don't know how to use them. For example, I want to use Map to display data and Legend to display content. I know it's not a good choice because DotSpatial is displays a map in a .NET Windows Forms, but for some reason I have to use it in WPF application, please help me.

Development Framework：
DotSpatial version:3.0.1.0
.NET Framework version:4.7.2

The picture below is the WindowsFormsApp I developed with .NET Framework. I want to implement it in WPF. Please tell me how to implement it.

![image](https://user-images.githubusercontent.com/61216968/231967895-e844d875-2068-4dc0-b37f-efd6edd7ba98.png)

**Comment by sindizzy** [comment]
They are two different systems.

**Comment by MatthiasSchilder** [comment]
You can use the Map and Legend control through a WindowsFormsHost in WPF. Works without problems for us.

**Comment by alsl521** [comment]
I tried what you guys said. I designed two panels: MapPanel.xaml and LegendPanel.xaml, and then displayed them in MainWindow.xaml. I can display the map normally, but I don't know how to use Legend. And I don't know how to link Map and Legend in this case. I hope you guys can help me if possible. I really appreciate you guys.

Development Framework：
DotSpatial version:3.0.1.0
.NET Framework version:4.7.2

I only write so much code below, I won't do the rest.

MapPanel.xaml

```c#
<WindowsFormsHost>
	<dot:Map x:Name="Mapcontrol"/>
</WindowsFormsHost>
```

MapPanel.xaml.cs

```C#
Mapcontrol.AddLayer("F://arcgis//20230415//20230415//xian.shp");
```

LegendPanel.xaml

```c#
<WindowsFormsHost>
	<dot:Legend x:Name="Legendcontrol"/>
</WindowsFormsHost>
```

MainWindow.xaml

```c#
<Grid ShowGridLines="True" >
	<Grid.ColumnDefinitions>
		<ColumnDefinition></ColumnDefinition>
		<ColumnDefinition></ColumnDefinition>
	</Grid.ColumnDefinitions>
	<test:LegendPanel Grid.Column="0"/>
	<test:MapPanel Grid.Column="1" />
</Grid>
```



![image](https://user-images.githubusercontent.com/61216968/232513613-cfa6b14c-4a01-440f-9401-a54943f2f052.png)
